lib/v2/ibc/radio.js

```diff
@@ -0,0 +1,72 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const timezone = require('@/utils/timezone');
+const { parseDate } = require('@/utils/parse-date');
+const { art } = require('@/utils/render');
+const path = require('path');
+
+module.exports = async (ctx) => {
+    const id = ctx.params.id ?? 'maitta';
+    const limit = ctx.query.limit ? parseInt(ctx.query.limit) : 150;
+
+    const rootUrl = 'https://www.ibc.co.jp';
+    const currentUrl = `${rootUrl}/radio/${id}/audio`;
+
+    const response = await got({
+        method: 'get',
+        url: currentUrl,
+    });
+
+    const $ = cheerio.load(response.data);
+
+    const image = $('meta[property="og:image"]').attr('content');
+
+    let items = $('.audio_list section')
+        .slice(0, limit)
+        .toArray()
+        .map((item) => {
+            item = $(item);
+
+            return {
+                title: item.find('h3').text(),
+                link: new URL(item.find('a').attr('href'), rootUrl).href,
+                pubDate: timezone(parseDate(item.find('.update_date').text(), 'YYYY年MM月DD日'), +9),
+            };
+        });
+
+    items = await Promise.all(
+        items.map((item) =>
+            ctx.cache.tryGet(item.link, async () => {
+                const detailResponse = await got({
+                    method: 'get',
+                    url: item.link,
+                });
+
+                const content = cheerio.load(detailResponse.data);
+
+                const audio = `https:${content('audio').attr('src').split('?')[0]}`;
```

Comment:
The audio URL seems follow a pattern of `media-data.cdn.ibc.co.jp/out/sound/${authorId}_${audioId}/${authorId}_${audioId}.mp3`. You can visit `items[0]` for `authorId` and create the rest using data from `currentUrl` instead of sending tons of requests to the server.

Reply:
👍